hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/TestStorageContainerManager.java

```diff
@@ -604,8 +602,13 @@ public void testCloseContainerCommandOnRestart() throws Exception {
       // Stop processing HB
       scm.getDatanodeProtocolServer().stop();
 
-      scm.getContainerManager().updateContainerState(selectedContainer
-          .containerID(), HddsProtos.LifeCycleEvent.FINALIZE);
+      LoggerFactory.getLogger(TestStorageContainerManager.class).info(
+          "Current Container State is" + selectedContainer.getState());
+      if (selectedContainer.getState() == HddsProtos.LifeCycleState.OPEN) {
+        scm.getContainerManager().updateContainerState(selectedContainer
+            .containerID(), HddsProtos.LifeCycleEvent.FINALIZE);
```

Comment:
Do you know what other thread changed container state to CLOSING?  Don't we still have a race condition here: in the time window between `getState` and `updateContainerState` that other thread could still change state of the container?

Reply:
no, I couldn't find the other thread closing the container. one theory could be, We stopped processing the heartbeat from the datanode and if it assumes data node is dead, it could close the container. 
Yes there is a race condition still unless it could be done atomically. I changed it to try/catch to avoid any race condition.